redux/apis/auth-api.ts

```diff
@@ -49,19 +50,28 @@ export const authApi = createApi({
       }),
     }),
 
-    verifyLink: builder.query<VerifyEmailResponse | any, string>({
+    verifyLink: builder.query<AuthenticationResponse | any, string>({
       query: (token) => ({
         method: "GET",
         url: "/auth/verify-link",
         params: { token },
       }),
       keepUnusedDataFor: 0.001,
     }),
+
+    signIn: builder.mutation<AuthenticationResponse, SignInRequest>({
```

Comment:
add error response interface handling as well

Reply:
@imrankhan-stixor, It doesn't make any sense to me. From my understanding `Return Type` refers to the data type that the query function is expected to return upon a successful API call. By using `builder.mutation<AuthenticationResponse | ErrorResponse, SignInRequest>`, we're specifying that the return type will either be AuthenticationResponse or ErrorResponse. This will cause confusion when accessing it on the frontend. Let me know if am missing anything.

Reply:
Okay if the error can be handled without the use of error type then we'll use this approach, just make sure if error andling is working and then resolve this one